beacon-chain/rpc/core/validator.go

```diff
@@ -209,3 +233,40 @@ func SubmitSignedContributionAndProof(
 
 	return nil
 }
+
+// SubmitSignedAggregateSelectionProof verifies given aggregate and proofs and publishes them on appropriate gossipsub topic.
+func SubmitSignedAggregateSelectionProof(
+	ctx context.Context,
+	req *ethpb.SignedAggregateSubmitRequest,
+	genesisTime time.Time,
+	broadcaster p2p.Broadcaster,
+) *RpcError {
+	if req.SignedAggregateAndProof == nil || req.SignedAggregateAndProof.Message == nil ||
```

Comment:
I think this is the only v1alpha1 endpoint with such extensive checks. I don't really care about this code too much because these fields are validated in the handler anyway, but I didn't want to change the logic.